its/src/test/java/com/sonar/it/csharp/RazorClassLibProjectTest.java

```diff
@@ -63,4 +64,10 @@ void projectIsAnalyzed() {
 
     assertThat(getComponent(RAZOR_COMPONENT_CLASS_FILE).getName()).isEqualTo("Component.razor");
   }
+  
+  @Test
+  void issueS6801IsRaised() {
+    var issuesS6801 = Tests.getIssues(PROJECT).stream().filter(x -> x.getRule().startsWith(SONAR_RULE_S6801)).collect(Collectors.toList());
+    assertThat(issuesS6801).hasSize(1);
```

Comment:
Nitpick: check that the issue is from a specific file:
```java
assertThat(issuesS6801.stream().filter(issue -> issue.getComponent().equals(S6801_CONSUMRE_COMPONENT_FILE))).hasSize(1);
```
Where `S6801_COMPONENT_FILE` is:

```java
private static final String S6801_CONSUMRE_COMPONENT_FILE = "RazorClassLib:S6801/Consumer.razor";
```